src/celeritas/field/DormandPrinceMultiStepperGlobal.cuda.hh

```diff
@@ -0,0 +1,280 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/field/DormandMultiPrinceStepperGlobal.cuda.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/math/Algorithms.hh"
+
+#include "Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Based on the DormandPrinceStepper.hh, but with multiple threads.
+ */
+template<class EquationT>
+class DormandPrinceMultiStepperGlobal
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using result_type = FieldStepperResult;
+    //!@}
+
+  public:
+    //! Construct with the equation of motion
+    explicit CELER_FUNCTION DormandPrinceMultiStepperGlobal(EquationT&& eq)
+        : calc_rhs_(::celeritas::forward<EquationT>(eq))
+    {
+    }
+
+    // Adaptive step size control
+    CELER_FUNCTION result_type operator()(real_type step,
+                                          OdeState const& beg_state,
+                                          int number_threads,
+                                          OdeState* ks,
+                                          OdeState* along_state,
+                                          FieldStepperResult* result) const;
+
+    CELER_FUNCTION void run_sequential(real_type step,
+                                       OdeState const& beg_state,
+                                       int id,
+                                       int mask,
+                                       OdeState* ks,
+                                       OdeState* along_state,
+                                       FieldStepperResult* result) const;
+
+    CELER_FUNCTION void run_aside(real_type step,
+                                  OdeState const& beg_state,
+                                  int id,
+                                  int index,
+                                  int mask,
+                                  OdeState* ks,
+                                  OdeState* along_state,
+                                  FieldStepperResult* result) const;
+
+  private:
+    // Functor to calculate the force applied to a particle
+    EquationT calc_rhs_;
+};
+
+//---------------------------------------------------------------------------//
+// MACROS
+//---------------------------------------------------------------------------//
+#define UPDATE_STATE(index, state, coefficient, k)        \
```

Comment:
undef macros once you no longer need them (at the end of the header). Also if you can use lambdas instead.